padl/transforms.py

```diff
@@ -2184,24 +2425,78 @@ class _ItemGetter:
     :param default: The default value to fall back to in case of exception.
     """
 
-    def __init__(self, samples, transform, exception=None, default=None):
+    def __init__(self, samples, transform, entire_transform, exception=None, default=None):
         self.samples = samples
         self.transform = transform
+        self.entire_transform = entire_transform
+        self.exception = exception
+        self.default = default
+
+    def __getitem__(self, item):
+        if self.exception:
```

Comment:
Is it possible to rewrite this block like this?
```python
try:
    return item, self.transform(self.samples[item])
except self.exception:
    return item, self.default
except Exception as err:
    is_entire_transform = str(self.entire_transform) == str(self.entire_transform.pd_preprocess)
    self.entire_transform._pd_trace_error(
        self.entire_transform.pd_preprocess._pd_get_target_stage_idx(is_entire_transform),
        [self.samples[item]]
    )
    raise err
```

Reply:
It looks like it's possible to handle multiple exceptions in python as shown here https://stackoverflow.com/questions/6095717/python-one-try-multiple-except

Reply:
The problem with that is that we are missing the `if self.exception` check, that allows us to make sure that `self.exception` is not `None` (the default value) and then we don't get the following error:
```
TypeError: catching classes that do not inherit from BaseException is not allowed
```
if we try to go into it.